src/build/_importlib.py

```diff
@@ -1,14 +1,13 @@
 import sys
 
 
-if sys.version_info < (3, 8):
-    import importlib_metadata as metadata
-elif sys.version_info < (3, 9, 10) or (3, 10, 0) <= sys.version_info < (3, 10, 2):
+if sys.version_info >= (3, 10, 2):
+    from importlib import metadata  # type: ignore[attr-defined]
```

Comment:
The `type: ignore` workaround is needed to pass the "type" job https://github.com/pypa/build/actions/runs/6558894339/job/17813388696?pr=693
I didn't find a better way 🤔

Reply:
Something like this?

```python
if sys.version_info < (3, 8):
    import importlib_metadata as metadata
elif sys.version_info >= (3, 10, 2):
    from importlib import metadata
else:
    try:
        import importlib_metadata as metadata
    except ModuleNotFoundError:
        # helps bootstrapping when dependencies aren't installed
        from importlib import metadata
```

Reply:
Yes this worked, thanks (side note, I'm surprised that mypy still raises an error when the the order of the first two conditions is inverted)